src/client/datascience/jupyter/debuggerVariables.ts

```diff
@@ -245,6 +271,16 @@ export class DebuggerVariables extends DebugLocationTracker
                 this.currentVariablesReference = newVariablesReference;
                 this.currentSeqNumsForVariables.clear();
             }
+
+            // Catch the scopes response and use its variablesReference to send a variables message
+            const doc = this.vscNotebook.activeNotebookEditor?.document;
+            if (this.activeNotebookIsDebugging() && doc) {
+                const callVariables = async () => {
+                    const session = await this.debuggingManager.getDebugSession(doc);
+                    void session?.customRequest('variables', { variablesReference: newVariablesReference });
```

Comment:
Would rather see all the requests that DebuggerVariables has to do lined up so I can read the actual flow in one place. For example it should look more like

```
await session.customRequest('variables', ...)
this.refreshEventEmitter()
```

Reply:
Ideally I guess you would make all the variables requests for all scopes, then only fire the refresh event a single time. Currently it will be fired multiple times to trigger multiple refreshes quickly